### PR TITLE
Error handling when container not exists

### PR DESCRIPTION
**Pull request by rajasec**
When the container not exists, user gets the error saying
open /run/runc/container1/state.json: no such file or directory
 From end user perspective, if we do any operation on container we can show whether the container ID is valid one, instead of giving the state.json error.

Hence the change in this PR , upon executing runc on invalid ID, it will show as below.
"container not exist"
Signed-off-by: rajasec <rajasec79@gmail.com>

### CODE REVIEW

File: libcontainer/factory_linux.go [hunk 1]
```diff
@@ -277,7 +277,7 @@ func (l *LinuxFactory) loadState(root string) (*State, error) {
 	f, err := os.Open(filepath.Join(root, stateFilename))
 	if err != nil {
 		if os.IsNotExist(err) {
-			return nil, newGenericError(err, ContainerNotExists)
+			return nil, newGenericError(fmt.Errorf("container not exists"), ContainerNotExists)
```
Comment (crosbymichael) [inline libcontainer/factory_linux.go]: Can you add the container's id to the error message? 

`"container %q does not exist"`

**Comment by rajasec** [comment]
Added the container ID as per your comment, since root was passed ( it contains the full path: /run/runc/<containerID) so instead of printing the full path or trimming down this path, I passed the id as argument via loadState to get the ID.

File: libcontainer/factory_linux.go [hunk 2]
```diff
@@ -273,11 +273,11 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 	return i.Init()
 }
 
-func (l *LinuxFactory) loadState(root string) (*State, error) {
+func (l *LinuxFactory) loadState(root, id string) (*State, error) {
 	f, err := os.Open(filepath.Join(root, stateFilename))
 	if err != nil {
 		if os.IsNotExist(err) {
-			return nil, newGenericError(err, ContainerNotExists)
+			return nil, newGenericError(fmt.Errorf("container id %q not exists", id), ContainerNotExists)
```
Comment (crosbymichael) [inline libcontainer/factory_linux.go]: Can you remove the "id" part.  It should just read `container %q does not exist`  To people they associate the id as the container.

**Comment by rajasec** [comment]
Updated as per your comment.

**Comment by crosbymichael** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**